### PR TITLE
fix: show biometrics prompt everytime `authenticate` is invoked for MacOS

### PR DESCRIPTION
**Pull request by hpelitebook745G2**
**Issue**: I encountered a behavior where the biometrics prompt appeared only during the initial authentication. Subsequent authentication attempts done shortly after, did not trigger the said prompt. Our specific use case requires the biometrics prompt to be displayed consistently to prevent misuse.

**Solution**:
Updated the `FlutterLocalAuthenticationPlugin.swift` and passed `context` to the respective methods instead of using it globally to persist `touchIDAuthenticationAllowableReuseDuration = 0`. This ensures that the biometrics prompt is displayed consistently whenever the user attempts to authenticate, regardless of the time elapsed since the previous authentication.

### CODE REVIEW

File: macos/Classes/FlutterLocalAuthenticationPlugin.swift [hunk 1]
```diff
@@ -3,23 +3,23 @@ import LocalAuthentication
...
-    
+
     public static func register(with registrar: FlutterPluginRegistrar) {
         let channel = FlutterMethodChannel(name: "flutter_local_authentication", binaryMessenger: registrar.messenger)
         let instance = FlutterLocalAuthenticationPlugin()
         registrar.addMethodCallDelegate(instance, channel: channel)
     }
 
     public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+        let context = LAContext()
```
Comment (eaceto) [inline macos/Classes/FlutterLocalAuthenticationPlugin.swift:14]: Hello, thanks very much for your contribution.

Recreating an authentication context each time this function is called is not a desired behaviour. If you want to prompt authentication each time, the *touchIDAuthenticationAllowableReuseDuration* should be configurable to something different than zero.

**Review by eaceto — COMMENTED** [review]
As commented inline in this PR, this introduces a side effect that's not desirable.

**Comment by eaceto** [comment]
Would you mind testing version 1.0.0 (published and available in main branch)? Thanks very much

**Comment by tmpfs** [comment]
@eaceto and @hpelitebook745G2 - I also want this behavior but reading the code and [checking the apple docs](https://developer.apple.com/documentation/localauthentication/lacontext/1622329-touchidauthenticationallowablere) this looks like a bug with Apple.

According to the Apple docs the default value is `0` which indicates that the authentication should not be re-used however I see a previous authentication being reused even when the value is zero.

The code in this plugin looks good to me.

**Comment by tmpfs** [comment]
Given that this is an Apple Bug as far as I can tell I wanted to understand what the cost of creating an `LAContext` is and ChatGPT had this to say:

> Creating an `LAContext` in macOS (or iOS) is not typically an expensive operation in terms of performance. `LAContext` is a lightweight object used for interacting with the Local Authentication framework, which includes features like Touch ID and Face ID.

> The `LAContext` object itself does not perform any actual authentication or validation until you make specific requests, such as fingerprint or facial recognition checks. These checks are the operations that can be computationally more expensive, as they involve cryptographic operations and hardware checks to ensure the security of biometric data.

> Creating an `LAContext` is relatively inexpensive because it's mainly an object for managing authentication requests, storing settings, and handling the communication between your app and the system framework. It doesn't perform any intensive tasks during its creation.

> The actual cost in terms of time and system resources comes when you call methods like `evaluatePolicy(_:localizedReason:reply:)` to perform authentication checks, as this is when the system performs the necessary computations and communicates with biometric hardware. These operations can vary in complexity and duration depending on the specific request and the device's capabilities.

> So, creating an `LAContext` should not be a significant performance concern on its own, but it's the actual biometric authentication checks that may have an impact on performance, especially if they are called frequently or if they involve a large amount of data. Therefore, it's essential to use these features judiciously and consider the performance implications when integrating biometric authentication into your macOS application.

So @eaceto if you are adamant that this is not a worthwhile workaround I am happy to maintain a fork with this fix otherwise let's work together to implement a workaround which addresses this problem.

Edit: if you do want to collaborate here is a commit that adds `allowReuse` to `authenticate()` (MacOS only):

https://github.com/saveoursecrets/flutter_local_authentication/commit/876a64521843610ee7853c12fa1211c20ad8eec5

**Comment by hpelitebook745G2** [comment]
> Would you mind testing version 1.0.0 (published and available in main branch)? Thanks very much

hi @eaceto , thanks for the updates!

`setTouchIDAuthenticationAllowableReuseDuration(0.0)` doesn't seem to work.

is there another way to make the biometrics prompt everytime i want it?

**Pull request closed without merging**